File: spring-boot-backend/src/main/java/com/eafc/springbootbackend/entities/shopping/OrderInfo.java
package com.eafc.springbootbackend.entities.shopping;

import com.eafc.springbootbackend.entities.customer.AccountInfo;
import lombok.Data;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.Calendar;
import java.util.Collection;

@Entity
@Data
public class OrderInfo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long orderId;

    @NotNull
    private Calendar purchaseDate;

    private boolean pending;

    @NotNull
    private double total;

//    @NotNull
    @ManyToOne
    @JoinColumn(name = "paymentMethod")
    private PaymentMethod paymentMethod;

    @ManyToOne
    @JoinColumn(name = "customer")
    private AccountInfo customer;
}

File: angular-frontend/src/app/common/users/orderInfo.ts
import {ProductInfo} from "../shopping/product-info";
import {AccountInfo} from "./account-info";
import {PaymentMethod} from "../shopping/payment-method";
import {ShippingAddress} from "./shipping-address";

export class OrderInfo {
  orderId!: number;
  purchaseDate!: string;
  pending!: boolean;
  total!: number;
  paymentMethod!: PaymentMethod;
  shippingAddress!: ShippingAddress;
}
